### PR TITLE
Make max length validation more strict

### PR DESCRIPTION
**Pull request by htdvisser**
#### Summary

This pull request addresses #3577 by adding max length validations on `string`, `bytes` fields, as well as `maps` and `repeated` fields.

#### Testing

For most fields I validated that our production databases do not contain any values that are even close to these limits.

##### Regressions

This could break consistency with validation rules in clients such as the Console.

#### Notes for Reviewers

@johanstokking please see if any additional limits need to be set in JS and EDTC API.

@rvolosatovs please see if any additional limits need to be set in NS (EndDevice) APIs.

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- ⚠️ Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
	- As with anything relating to API this could break clients that rely on buggy (non validated) behavior.
- ⚠️ Documentation: Relevant documentation is added or updated.
	- This will be updated the next time we sync API descriptors to the documentation site
- [x] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

**Comment by htdvisser** [comment]
@kschiffer or @bafonins: please take a look and see if any changes are needed in the Console as a result of these changes. If not, please comment and remove your review request.

File: api/gateway.proto [hunk 1]
```diff
@@ -244,7 +259,13 @@ message GatewayAntenna {
   float gain = 1;
   // location is the antenna's location.
   Location location = 2 [(gogoproto.nullable) = false, (validate.rules).message.required = true];
-  map<string,string> attributes = 3 [(validate.rules).map.keys.string = {pattern: "^[a-z0-9](?:[-]?[a-z0-9]){2,}$" , max_len: 36}];
+  map<string,string> attributes = 3 [
+    (validate.rules).map = {
+      max_pairs: 10,
```
Comment (kschiffer) [inline api/gateway.proto:271]: Why only 10?

Comment (htdvisser) [reply api/gateway.proto:271]: Like any other limit we set, it's pretty arbitrary. In our production databases we don't see any entities with more than 10 attributes. Most entities don't have any attributes. Entities that do have attributes, have between 1 and 6. So let's turn the question around: Why more than 10?

Comment (kschiffer) [reply api/gateway.proto:271]: I guess for how we use attributes currently it's fine. I remember some discussion around using labels e.g. for tags, in which case I can see more than 10 attributes being beneficial. As long as we can increase the limit later on, it's fine I'd say.

**Comment by kschiffer** [comment]
I went through and pushed some validation rule updates accordingly.

**Review by rvolosatovs — APPROVED** [review]

**Review by bafonins — APPROVED** [review]

**Comment by htdvisser** [comment]
I'll merge this with a failing Go test, because that one comes from https://github.com/TheThingsNetwork/lorawan-stack/pull/3679

**Pull request merged**